### PR TITLE
Add dependabot golang updates to Dockerfile

### PR DESCRIPTION
**Pull request by Kostov6**
**How to categorize this PR?**

/kind enhancement
/os garden-linux

**What this PR does / why we need it**:
Dependabot will now open PRs for golang version update in Dockerfile

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @Kostov6 for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -8,4 +8,9 @@ updates:
   allow:
   - dependency-name: "github.com/gardener/gardener"
   labels:
```
Comment (ialidzhikov) [inline .github/dependabot.yaml:10]: As we discussed, let's remove the label if it is not causing any troubles for gardener-robot. dependabot already adds its own labels such as `dependencies` and `go`.

**Review by ialidzhikov — COMMENTED** [review]
Nits, otherwise lgtm

**Review by ialidzhikov — COMMENTED** [review]
Nits, otherwise lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**